openpathsampling/engines/external_snapshots/snapshot.py

```diff
@@ -106,10 +106,9 @@ def set_details(self, xyz, velocities, box_vectors):
             pass
         else:
             raise RuntimeError("Can't set details if frame already exists.")
-        finally:
-            self._xyz = xyz
-            self._velocities = velocities
-            self._box_vectors = box_vectors
+        self._xyz = xyz
+        self._velocities = velocities
+        self._box_vectors = box_vectors
```

Comment:
This is probably better to be put in the `except` clause (cant suggest becuase this involves unchanged lines) (and the except should probably catch a *very* specific error

Reply:
Moved this into the `except` clause in ebb21e9. I didn't specify the exception because I'm not entirely sure that we can predict what the exception should be (depends on the implementation of `engine.read_frame_data`).